modules/core/src/main/java/org/apache/ignite/internal/processors/metric/GridMetricManager.java

```diff
@@ -234,6 +241,9 @@
     /** Nonheap memory metrics. */
     private final MemoryUsageMetrics nonHeap;
 
+    /** Profiling. */
+    private final FileProfiling profiling;
```

Comment:
It seems, code that related to performance statistics should be moved to separate Manager.

Reply:
I have moved code to the `PerformaceStatisticsProcessor`